MHVLib/HVMobile/HealthVaultRecord.m

```diff
@@ -76,30 +76,18 @@ - (id)initWithXml: (NSString *)xml
 
 
 - (BOOL)parseFromXml: (NSString *)xml {
-
-    @autoreleasepool
-    {
-        @try {
-            
-            XmlTextReader *xmlReader = [XmlTextReader new];
-            XmlElement *root = [xmlReader read: xml];
-            
-            if (!root) {
-                return NO;
-            }
-            
-            self.recordId = [root attrValue: @"id"];
-            self.recordName = [root text];
-            self.authStatus = [root attrValue: @"app-record-auth-action"];
-            
-        }
-        @catch (id exc) {
-            
-            return NO;
-        }
-	}
-
-	return YES;
+    XReader *reader = [[XReader alloc] initFromString:xml];
+    [reader readStartElementWithName:@"record"];
+    
+    self.recordName = [reader readString];
+    self.recordId = [reader readAttribute:@"id"];
+    self.authStatus = [reader readAttribute:@"app-record-auth-action"];
+    
+    if (!self.recordId) {
+        return NO;
+    } else {
+        return YES;
+    }
```

Comment:
While we're modifying this file, lets make the code style consistent - move { to the next line in all cases. Also, it should be safe to remove the iVar definitions in the header file, the private interface definition, and the synthesize statements.